apps/devhub/tests/test_tasks.py

```diff
@@ -127,12 +131,34 @@ class TestValidator(amo.tests.TestCase):
 
     def setUp(self):
         super(TestValidator, self).setUp()
-        self.upload = FileUpload.objects.create()
+        self.upload = FileUpload.objects.create(
+            path=os.path.join(ADDON_TEST_FILES, 'desktop.xpi'))
         assert not self.upload.valid
 
     def get_upload(self):
         return FileUpload.objects.get(pk=self.upload.pk)
 
+    @contextmanager
+    def statsd_timing_mock(self):
+        statsd_calls = {}
+
+        def capture_timing_call(metric, value):
+            statsd_calls[metric] = value
+
+        with mock.patch('devhub.tasks.statsd.timing') as mock_timing:
+            mock_timing.side_effect = capture_timing_call
+            yield statsd_calls
+
+    def approximate_upload_time(self):
+        upload_start = utc_millesecs_from_epoch(self.upload.created)
+        now = utc_millesecs_from_epoch()
+        return now - upload_start
+
+    def assert_milleseconds_are_close(self, actual_ms, calculated_ms,
+                                      fuzz=Decimal(1000)):
```

Comment:
Heh, true. I am so pessimistic about Travis that I was giving it one second to run the ticks necessary to get from the end of the task handler to the assertion in the test code. That's probably unnecessarily long.